CMakeLists.txt

```diff
@@ -60,16 +60,28 @@ option( DETRAY_SVG_DISPLAY "Build ActSVG display module" OFF)
 option( DETRAY_BUILD_SYCL "Build the SYCL sources included in detray" OFF )
 option( DETRAY_BUILD_CUDA "Build the CUDA sources included in detray"
    ${DETRAY_BUILD_CUDA_DEFAULT} )
-option( DETRAY_BUILD_TESTING "Build the (unit) tests of Detray"
-   TRUE )
-cmake_dependent_option( DETRAY_BENCHMARKS "Enable benchmark tests" TRUE
-   "DETRAY_BUILD_TESTING" OFF )
-option( DETRAY_BUILD_TUTORIALS "Build the tutorial executables of Detray"
+option( DETRAY_BUILD_TEST_UTILS "Build the test utility library of Detray"
    ON )
+option( DETRAY_BUILD_UNITTESTS "Build the unit tests of Detray" ON )
+option( DETRAY_BUILD_INTEGRATIONTESTS "Build the integration tests of Detray" 
+   ON )
+option( DETRAY_BUILD_BENCHMARKS "Build the benchmark tests" ON  )
+option( DETRAY_BUILD_TUTORIALS "Build the tutorial executables of Detray" ON )
 option( DETRAY_FAIL_ON_WARNINGS
    "Make the build fail on compiler/linker warnings" FALSE )
 
-if ( ( BUILD_TESTING AND DETRAY_BUILD_TESTING ) OR DETRAY_BUILD_TUTORIALS )
+# Alias flag to build the tests
+if ( BUILD_TESTING OR DETRAY_BUILD_UNITTESTS OR DETRAY_BUILD_INTEGRATIONTESTS )
+   set( DETRAY_BUILD_TESTING ON )
+endif()
+
+# Test utilities are needed for tests, tutorials and benchmarks
+if ( DETRAY_BUILD_TESTING OR DETRAY_BUILD_BENCHMARKS OR DETRAY_BUILD_TUTORIALS )
+   set( DETRAY_BUILD_TEST_UTILS ON )
+endif()
```

Comment:
Does it make sense for the tutorials to depend on test utils?

Reply:
How to use some of those test utils, e.g. the ray scan or the toy detector, is shown in the tutorials

Reply:
Should these kind of dependencies be resolved in cmake or by the user?

Reply:
I think turning things _ON_ silently should be allowed on the CMake level to allow it to do what the user wants. We should not turn things _OFF_ silently though!